packages/x-data-grid-generator/src/hooks/useQuery.ts

```diff
@@ -146,9 +146,11 @@ export const loadServerRows = (
     firstRowIndex = page * pageSize;
     lastRowIndex = (page + 1) * pageSize;
   }
+  const hasNextPage = lastRowIndex < filteredRows.length;
```

Comment:
```suggestion
  const hasNextPage = lastRowIndex < filteredRows.length - 1;
```
Since index is 0-based.